### PR TITLE
Use Github Actions to run the tests and remove Travis

### PR DESCRIPTION
**Pull request by jpichon**
I'm not 100% about the Mergify change so we may want to be careful about approving before CI passes at first. From https://github.com/Mergifyio/mergify-engine/discussions/1995 the job names need to be spelled out for the entire matrix.

### CODE REVIEW

File: .github/workflows/test.yml
Comment (jguiditta) [inline .github/workflows/test.yml:33]: So we don't want to run flake the matrix of versions?

File: .github/workflows/test.yml
Comment (jguiditta) [inline .github/workflows/test.yml:26]: I think you probably know this, but just to be sure - I believe this sets up a separate container to run flake8.  This may not be an issue, but it is likely to make the tests run a bit slower.  I believe on the ones I changed, I just made flake8 a step, which would also run it for each of the pythons listed above.

**Comment by fuzzball81** [comment]
We will also need to change the mergify config to not use travis anymore. Not sure if it has github action support or not.

**Comment by fuzzball81** [comment]
Sigh it helps if I read the file names while looking at the changes. I see the mergify changes now.

File: .mergify.yml
Comment (jguiditta) [inline .mergify.yml:10]: I think you should be able to just make this one line, using:
```
- status-success=Run tests
```

I do not believe you need to specify each job. I saw the linked mergify question, and the maintainer's response can be interpreted different ways.  I think while the api may _return_ all of these names, you can still specify the top-level name.  I am ok with leaving this for now though, we can always experiment (and check the api, I haven't found exactly where to look yet) and simplify it later.

File: tox.ini
Comment (jguiditta) [inline tox.ini:4]: Perhaps this is not needed, but when I did this, I changed travis to gh-actions here (could just be a different way and both work, not sure).  See here:
https://github.com/release-depot/flask-container-scaffold/blob/main/tox.ini

**Review by jguiditta — COMMENTED** [review]
Overall, looks good, couple comments/thoughts.

File: .github/workflows/test.yml
Comment (jpichon) [reply .github/workflows/test.yml:33]: This doesn't seem useful for flake8 since linting isn't version dependent?

File: .github/workflows/test.yml
Comment (jpichon) [reply .github/workflows/test.yml:26]: I did this on purpose indeed, since it seems slower and unnecessary to me to repeat the flake8 step for every version?

File: tox.ini
Comment (jpichon) [reply tox.ini:4]: I tested the job against my branch while working on it and it seems to set up the versions correctly without it: https://github.com/jpichon/git_wrapper/actions/runs/574993890

File: .mergify.yml
Comment (jpichon) [reply .mergify.yml:10]: I'd love to have this be just one step, it would be clearer, less unwieldy and we wouldn't need to remember to change it with every python version update. I read the maintainer answer as requiring every job name to be exact in the mergify file. I'm not sure about the best way to go about testing this either, suggestions welcome!

File: tox.ini
Comment (jpichon) [reply tox.ini:4]: Then again, that might be why I had to specify -epy: just using "tox" as a command ran zero tests.

Thinking further, I'm tempted to leave it as is because it means updating the version matrix only requires a change in the workflow file, as opposed to remembering to update both test.yml and tox.ini?

File: .mergify.yml
Comment (jpichon) [reply .mergify.yml:10]: I compressed it into one line - let's be careful with the next MR and see if that works!

File: .github/workflows/test.yml
Comment (jguiditta) [reply .github/workflows/test.yml:33]: Sure, I can see that

File: .github/workflows/test.yml
Comment (jguiditta) [reply .github/workflows/test.yml:26]: Also fair

File: .mergify.yml
Comment (jguiditta) [reply .mergify.yml:10]: Cool - keep in mind we won't really know until we merge the PR with the mergify change (so meta!).  So it is possible we will need a rev or two if this doesn't work, but at least we know the actions are godd, since we can see how they ran on your branch when you pushed the commit(s)

File: tox.ini
Comment (jguiditta) [reply tox.ini:4]: I can see that as a benefit, and have no issue with it - perhaps we will change the other projects to do the same.

**Review by jguiditta — APPROVED** [review]
Actions pass, and as I said in another comment, I think we can rev mergify if it doesn't work as expected.

**Comment by jpichon** [comment]
Looks like the Travis reporting hung again, to highlight how necessary this patch is? ;)

**Comment by jguiditta** [comment]
> Looks like the Travis reporting hung again, to highlight how necessary this patch is? ;)

Of course! I'll manually merge it now, then we can forget travis

**Comment by jpichon** [comment]
Thanks! And now I realised I kept "main" instead of "master" in the branch names so it won't work at all......... I'll push an update.

**Pull request merged**
